server/middlewares.go

```diff
@@ -45,3 +46,22 @@ func serverTimeoutMW(initCtx context.Context) endpoint.Middleware {
 		}
 	}
 }
+
+// newCtxInjectMW must be placed at the end
+func newCtxInjectMW() endpoint.Middleware {
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, request, response interface{}) (err error) {
+			args, ok := request.(*streaming.Args)
+			if !ok {
+				return next(ctx, request, response)
+			}
+			// use contextStream to wrap the original Stream and rewrite Context()
```

Comment:
no need since we have solved the issue at `HandleStreams`? or the logic is redundant.